### ISSUE TITLE
the process of self-supervised experiment

### ISSUE THREAD

**Issue by changruguo** [emadeldeen24/TS-TCC#24]
hello. I have read your paper and code. In your code, I am so confused about the experiment process of fine-tune for self-supervised model. The following is my understanding: First you trained the pre-trained model using train.pt data. Then, in fine-tune, copy the pretrained model parameter in model, the next is my cofused: why do you conduct supervised training using train.pt again? why are the parameters of the pre-trained model not frozen?  In your abstract, you write you propose  an unsupervised method, Where does unsupervision manifest itself? In my understanding, in fine-tune process, you should forzen the pre-trained model parameter, and using fewer labeled data to finetune the classifier after the pretrained model. 

Looking forward to your answer, thank you.

**Comment by emadeldeen24** [comment]
You are confused about two different experiments:

1. fine-tuning, where the pretrained model weights are not frozen, but 'fine-tuned' according to the downstream task.
2. Linear probing, where the pretrained model weights are frozen to evaluate their efficacy. Only the classifier is trained in this experiment.

Therefore, fine-tuning can be the same as supervised training, but with loading the pretrained model first. You can also reduce the learning rate for the pretrained part of the model. 

Fine-tuning can be done with any proportion of labels available (despite its impact really appears with less data). I used fractions of data in my experiments. In the code, I load `train.pt`, but you can replace it with fractions of data to evaluate. You can check [this snippet](https://github.com/emadeldeen24/eval_ssl_ssc/blob/main/split_k-fold_and_few_labels.py) to create less data.

> Where does unsupervision manifest itself

In the unsupervised method, you train the model with the samples only without their labels.

**Comment by changruguo** [comment]
thank you for your reply
I have understood the finetune and train_linear process.
As for the unsupersion, because in your code, the number of labelled data you used is the same as the supvised method, so I  am confused it, and you give me the answer that any rate of finetune dataset is ok.
now, I have an immature idea, I split the dataset with 60% for train, 20% for finetune, and 20% for test. After pretrained the model with train dataset, then frozen the parameter, then using 20% finetune dataset train the linear classifier. 
what do you think of this idea?
In other words,
For the single dataset, How do you think you might design a self-supervised experiment? 
For pretrained model transfering(pretrained on a dataset, finetune and test on another dataset), How do you think you might design a self-supervised experiment? 

Looking forward to your reply, thank you.

**Comment by emadeldeen24** [comment]
If your objective is to improve the performance on one dataset, you can use the training set for self-supervised pretraining (without labels), and then use it again for fine-tuning. The validation set can be used to select the best hyperparameters for SSL and FT tasks. Later, you test on the test set.

If your objective is improving cross-dataset performance (say you want to pretrain on dataset A to get the best performance on dataset B), then you probably do not need a test set for A. In this case, you can pretrain on the whole dataset A, then split dataset B into val-B and test-B. You can use the val-B to tune the hyperparameters and test on test-B.

In overall, this design can change based on your objectives.

**Comment by changruguo** [comment]
![image](https://github.com/emadeldeen24/TS-TCC/assets/114247110/d7674258-d425-4fa8-8684-4c482a8e83fc)
OK, thanks for your replying
and another question
the red circle i write might not corresponding to the paper code, and i think it should be delete?
I think divide N at the end is not nessaryf? because the loss you caculate is mean loss, just like your code

**Comment by emadeldeen24** [comment]
I updated the definition in the CA-TCC paper. You can check.
